src/Services/PaymentCreation.php

```diff
@@ -133,6 +134,26 @@ public function createPayment($mopId, ResponseAbstract $response, Basket $basket
             $payment->receivedAt = date('Y-m-d H:i:s');
         }
 
+        try {
+            /** @var CurrencyExchangeRepositoryContract $currencyService */
+            $currencyService = pluginApp(CurrencyExchangeRepositoryContract::class);
+
+            $defaultCurrency = $currencyService->getDefaultCurrency();
+
+            if ($payment->currency != $defaultCurrency) {
+                $payment->exchangeRatio = $currencyService->getExchangeRatioByCurrency($payment->currency);
```

Comment:
If `$payment->currency != $defaultCurrency` is true, then we have a foreign currency. In this case the exchange rate should be already set (in the basket or during order creation, if we have an order). Is it necessary to change this value? Maybe you should first check it:
```php
if ($payment->exchangeRatio <= 0.0) {
    $payment->exchangeRatio = $currencyService->getExchangeRatioByCurrency($payment->currency);
}
```

Reply:
I can't find any exchange ratio in the basket. The main problem here is that the payment is saved with the exchange ration = 1 if this code is not added. I also researched this on other payment methods and seems to be the right way.